File: java/0053.最大子数组和.java
class Solution {
    // [题解]1. O(n) t:1ms(100%) O(1) m:50.2M(43.4%) 动态规划
    public int maxSubArray(int[] nums) {
        int pre = 0, ans = nums[0];
        for (int num : nums) {
            pre = Math.max(pre + num, num);
            ans = Math.max(ans, pre);
        }
        return ans;
    }
}

File: python/0053.最大子数组和.py
class Solution:
    # [题解]1. O(n) t:176ms(45%) O(1) m:25.6M(62%) 动态规划
    def maxSubArray(self, nums: List[int]) -> int:
        pre = 0
        ans = nums[0]
        for num in nums:
            pre = max(pre + num, num)
            ans = max(pre, ans)
        return ans